src/segmentmytif/utils/io.py

```diff
@@ -2,24 +2,18 @@
 from pathlib import Path
 from typing import Any, Union
 
-import numpy as np
-import rasterio
+from rioxarray import rioxarray
 
 from segmentmytif.logging_config import log_array
 
 logger = logging.getLogger(__name__)
 
 
-def read_geotiff(input_path: Path) -> (np.ndarray, Any):
-    with rasterio.open(input_path) as src:
-        data = src.read()
-        profile = src.profile
-    log_array(data, logger, array_name=input_path)
-    return data, profile
+def read_geotiff(raster_path: Path) -> rioxarray.raster_array:
```

Comment:
Given we have this wrapping function, I would suggest moving `_set_compute_mode` from `main.py` to `io.py` and call it in `read_geotiff`. `read_geotiff` can take `compute_mode` and `chunks` as input areguments

Reply:
good idea. Let's do this later